### PR TITLE
Add OpenDream to unit testing

### PR DESCRIPTION
**Pull request by out-of-phaze**
## Description of changes
Ports OpenDream unit testing from tgstation/Goonstation/Paradise/etc.

## Why and what will this PR improve
OpenDream's parsing effectively acts as a more advanced linter, and in the future we'll be able to annotate return types on things. Presently, BYOND accepts but ignores those type annotations, but OD has a PR open (that I've been contributing to) to check them.

Also, once we do this, OD will add us to their CI, ensuring better compatibility.

## Authorship
Took `dm_annotate` from tgstation, and adapted parts of the workflow from them as well. Cross-referenced with Goonstation's implementation. I improved some aspects of it like passing `--version` using the BYOND version environment variables.

### CODE REVIEW

**Comment by ike709** [comment]
Pragma documentation: https://github.com/OpenDreamProject/OpenDream/wiki/Pragmas---Error-Emissions

**Comment by out-of-phaze** [comment]
Waiting on https://github.com/OpenDreamProject/OpenDream/issues/1754 now. Still open to feedback.

**Comment by out-of-phaze** [comment]
Ready to go after https://github.com/OpenDreamProject/OpenDream/pull/1762 is merged to OpenDream.

File: code/datums/mind/mind.dm [hunk 1]
```diff
@@ -399,6 +399,7 @@
 				H.implant_loyalty(H, override = TRUE)
 				log_admin("[key_name_admin(usr)] has loyalty implanted [current].")
 			else
+				pass()
```
Comment (MistakeNot4892) [inline code/datums/mind/mind.dm:402]: What is the point of this compared to just deleting the else?

File: code/modules/client/client_procs.dm [hunk 2]
```diff
@@ -125,6 +125,7 @@ var/global/list/localhost_addresses = list(
 
 	switch (connection)
 		if ("seeker", "web") // check for invalid connection type. do nothing if valid
+			pass()
```
Comment (MistakeNot4892) [inline code/modules/client/client_procs.dm:128]: Wouldn't this be cleaner if the switch was inverted and it just returned instead of this structure?

**Review by MistakeNot4892 — APPROVED** [review]

File: code/datums/mind/mind.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/datums/mind/mind.dm:402]: spacemandmm sometimes throws inexhaustive switch errors for certain things like rand() in a switch condition and *maybe* alert() and a few other things. in this case specifically it also makes it clear that the case was explicitly expected/accounted for by whoever wrote the code. i also just wanted to keep the structure of existing code, though i mean, i can easily remove it

Comment (MistakeNot4892) [reply code/datums/mind/mind.dm:402]: Doesn't impact much either way, just looked odd.

File: code/modules/client/client_procs.dm (hunk 2 above)
Comment (out-of-phaze) [reply code/modules/client/client_procs.dm:128]: what do you mean? would have to do `if(connection == "seeker" || connection == "web")` which doesn't feel any cleaner to me

Comment (MistakeNot4892) [reply code/modules/client/client_procs.dm:128]: It wouldn't need a switch, else or pass(), which imo would be cleaner. Not really fussed.

**Pull request merged**